File: examples/python/08_switch_providers.py
"""The same agent on different providers. Only the model string changes."""

import asyncio
import os

from openharness import Agent, calculator, run

MODELS = [
    ("ANTHROPIC_API_KEY", "anthropic:claude-sonnet-5"),
    ("OPENAI_API_KEY", "openai:gpt-6-luna"),
    ("GEMINI_API_KEY", "gemini:gemini-3.8-flash"),
    (None, "ollama:llama3.2"),  # local, no key
]

agent = Agent(instructions="Answer in one sentence. Use the calculator for math.", tools=[calculator])


async def main() -> None:
    for key, model in MODELS:
        if key and not os.environ.get(key):
            continue
        try:
            result = await run(agent, "What is 2**20 divided by 7?", model=model)
            print(f"{model:32} {result.output}")
        except Exception as e:  # e.g. Ollama not running
            print(f"{model:32} skipped: {e}")


asyncio.run(main())

File: examples/typescript/08_switch_providers.ts
// The same agent on different providers. Only the model string changes.
import { Agent, calculator, run } from "openharness";

const MODELS: [string | null, string][] = [
  ["ANTHROPIC_API_KEY", "anthropic:claude-sonnet-5"],
  ["OPENAI_API_KEY", "openai:gpt-6-luna"],
  ["GEMINI_API_KEY", "gemini:gemini-3.8-flash"],
  [null, "ollama:llama3.2"], // local, no key
];

const agent = new Agent({ instructions: "Answer in one sentence. Use the calculator for math.", tools: [calculator] });
for (const [key, model] of MODELS) {
  if (key && !process.env[key]) continue;
  try {
    const result = await run(agent, "What is 2**20 divided by 7?", { model });
    console.log(model.padEnd(32), result.output);
  } catch (e) {
    console.log(model.padEnd(32), `skipped: ${(e as Error).message}`);
  }
}
